src/core/src/service.rs

```diff
@@ -171,46 +169,41 @@ impl Poet2Service {
             .expect("Failed to send block ack");
     }
 
-    // Calculate the time to wait between publishing blocks. This will be a
-    // random number between the settings sawtooth.consensus.min_wait_time and
-    // sawtooth.consensus.max_wait_time if max > min, else DEFAULT_WAIT_TIME. If
-    // there is an error parsing those settings, the time will be
-    // DEFAULT_WAIT_TIME.
-    pub fn calculate_wait_time(&mut self, chain_head_id: BlockId) -> time::Duration {
-        let settings_result = self.service.get_settings(
-            chain_head_id,
-            vec![
-                String::from("sawtooth.consensus.min_wait_time"),
-                String::from("sawtooth.consensus.max_wait_time"),
-            ],
-        );
-
-        let wait_time = if let Ok(settings) = settings_result {
-            let ints: Vec<u64> = vec![
-                settings.get("sawtooth.consensus.min_wait_time").unwrap(),
-                settings.get("sawtooth.consensus.max_wait_time").unwrap(),
-            ].iter()
-                .map(|string| string.parse::<u64>())
-                .map(|result| result.unwrap_or(0))
-                .collect();
+    pub fn get_wait_time(&mut self, chain_head: Block, validator_id: &Vec<u8>) -> u64 // chain_head.block_id.clone() 
+    {
+        let mut duration64: u64 = 0_u64;
+        let mut prev_wait_certificate = String::new();
+        let mut prev_wait_certificate_sig = String::new();
 
-            let min_wait_time: u64 = ints[0];
-            let max_wait_time: u64 = ints[1];
-
-            debug!("Min: {:?} -- Max: {:?}", min_wait_time, max_wait_time);
-
-            if min_wait_time >= max_wait_time {
-                DEFAULT_WAIT_TIME
-            } else {
-                rand::thread_rng().gen_range(min_wait_time, max_wait_time)
-            }
+        debug!("Getting new wait time for next block.");
+        if chain_head.block_num == 0_u64 { // genesis block
+            duration64 = enclave::initialize_wait_certificate(
+                                 prev_wait_certificate,
+				 poet2_util::blockid_to_hex_string(chain_head.previous_id),
+				 prev_wait_certificate_sig,
+                                 &validator_id);
         } else {
-            DEFAULT_WAIT_TIME
-        };
-
-        info!("Wait time: {:?}", wait_time);
+            let payload = chain_head.payload.clone();
+            let delim_index = payload.iter().position(|&i| i == 0x23).unwrap(); // # char
```

Comment:
Put this 0x23 in some constant.

Reply:
Yes, this is happening in next iteration.